src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -75,19 +80,22 @@ public String toString() {
     public static class DescriptorImpl extends Descriptor<UserRemoteConfig> {
 
         public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project,
```

Comment:
well in the case of these `@WebMethod` signature changes, where it is stapler that binds and thus is flexible for the additional parameters, there can be an advantage as otherwise you would end up with two `doFillCredentialsIdItems` and then you don't know which one Stapler will invoke.

So there is advantage in adding it for the `doFill...` and `doCheck...` methods...

But as I said before, adding them to one method and not all the others is IMHO outside the scope of this PR which is why I didn't add the annotations.

I find it can be good to have them, but I wouldn't add them in a PR that does anything other than add `@Restricted(NoExternalUse.class) @RestrictedSince("...")` to all the methods that you want to have flexibility to evolve the signatures of